### PR TITLE
Fixes #37983 - Remove syspurpose addons from Katello

### PR DESCRIPTION
**Pull request by chris1984**
#### What are the changes introduced in this pull request?
* There are no longer any SKUs that use syspurpose addons. Additionally, the addons command is being removed from subscription-manager in RHEL10. So we are free to remove addons from Katello
* Removed systempurpose_addons from AngularJS, React, and the Rails backend
* Created a DB Migration to remove the addon tables

#### What are the testing steps for this pull request?
* Checkout PR and apply migration, make sure tables are removed
* Look around in the UI to make sure nothing broke
* Look around in the code to see if I missed anything
* Make sure the system porpoise dolphin did not sneak in 🥚

### CODE REVIEW

**Comment by parthaa** [comment]
can you rebase this ?

**Comment by chris1984** [comment]
> can you rebase this ?

Will you review it if I do :) I don't want to play the rebase game if no one is going to review it.

**Comment by chris1984** [comment]
> can you rebase this ?

@parthaa updated

File: db/migrate/20241101144625_remove_system_purpose_addons.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class RemoveSystemPurposeAddons < ActiveRecord::Migration[6.1]
+  def up
+    remove_foreign_key :katello_subscription_facet_purpose_addons, :katello_purpose_addons, column: :purpose_addon_id
+
+    drop_table :katello_activation_key_purpose_addons
+    drop_table :katello_purpose_addons
+    drop_table :katello_subscription_facet_purpose_addons
```
Comment (jeremylenz) [inline db/migrate/20241101144625_remove_system_purpose_addons.rb:7]: @parthaa Is this going to create problems with https://github.com/chris1984/katello/blob/sysporpose/db/migrate/20190605014649_add_purpose_addons.rb ?

do we need to do the "FakePurposeAddon" thing in that old migration file?

**Review by jeremylenz — COMMENTED** [review]
I am not able to run the down migration:
```
$ bundle exec rails db:migrate:down VERSION=20241101144625

== 20241101144625 RemoveSystemPurposeAddons: reverting ========================
-- create_table(:katello_activation_key_purpose_addons)
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

you can't redefine the primary key column 'id'. To define a custom primary key, pass { id: false } to create_table.
```

One option would be just to consider this an irreversible migration, delete the `down` method, and rename `up` to `change`.

Other than that everything seems to work as expected. :+1:

**Comment by chris1984** [comment]
> I am not able to run the down migration:
> 
> ```
> $ bundle exec rails db:migrate:down VERSION=20241101144625
> 
> == 20241101144625 RemoveSystemPurposeAddons: reverting ========================
> -- create_table(:katello_activation_key_purpose_addons)
> rails aborted!
> StandardError: An error has occurred, this and all later migrations canceled:
> 
> you can't redefine the primary key column 'id'. To define a custom primary key, pass { id: false } to create_table.
> ```
> 
> One option would be just to consider this an irreversible migration, delete the `down` method, and rename `up` to `change`.
> 
> Other than that everything seems to work as expected. 👍

Since I don't think we are going to be adding these back, might as well as change it to irreversible. I was debating on that but decided to try and be fancy and it blew up.

**Comment by chris1984** [comment]
@jeremylenz updated the migration

**Review by jeremylenz — CHANGES_REQUESTED** [review]
Clicking request changes until the https://github.com/Katello/katello/pull/11196/files#r1847225444 is resolved.

**Comment by chris1984** [comment]
> Clicking request changes until the https://github.com/Katello/katello/pull/11196/files#r1847225444 is resolved.

Talked with @parthaa and he had me update that migration to just remove us looping through things and just create the table.

**Review by jeremylenz — APPROVED** [review]

**Comment by chris1984** [comment]
Tested on a downstream Satellite with a `--reset` to test the database migration and it passed
```bash
2024-11-19 15:40:27 [INFO  ] [post] Executing hooks in group post
  Success!
  * Satellite is running at https://server.example.com

  * To install an additional Capsule on separate machine continue by running:

      capsule-certs-generate --foreman-proxy-fqdn "$CAPSULE" --certs-tar "/root/$CAPSULE-certs.tar"
  * Capsule is running at https://server.example.com

The full log is at /var/log/foreman-installer/satellite.log
Package versions are being locked.
2024-11-19 15:40:27 [INFO  ] [root] Package versions are being locked.
2024-11-19 15:40:28 [INFO  ] [post] All hooks in group post finished
```

**Comment by ekohl** [comment]
I see system purpose is still mentioned in the docs. In https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#Managing_Activation_Keys_content-management for example. Is this work tracked?

**Comment by jeremylenz** [comment]
System purpose addons is only one small part of system purpose. Other syspurpose attributes, (role, usage, SLA) along with release version, are neither deprecated nor removed.

**Comment by ekohl** [comment]
Yet it is mentioned (though the OracleLinux part is a false positive):
```console
 rg -i addons
guides/common/assembly_importing-content.adoc
58::yum_repo_extras_name: {os_name} {os_major} Addons
59::yum_repo_extras_url: https://yum.oracle.com/repo/OracleLinux/OL{os_major}/addons/x86_64/
70::yum_repo_extras_name: {os_name} {os_major} Addons
71::yum_repo_extras_url: https://yum.oracle.com/repo/OracleLinux/OL{os_major}/addons/x86_64/

guides/common/modules/proc_editing-the-system-purpose-of-a-host.adoc
31:# subscription-manager syspurpose add addons '_your_addon_'

guides/common/modules/proc_creating-an-activation-key.adoc
51:--purpose-addons "_addons_"
```

At least the last one is a hammer command and I think that will break.

**Comment by chris1984** [comment]
@ekohl I will make a PR to remove it from the docs today

**Comment by chris1984** [comment]
Docs PR https://github.com/theforeman/foreman-documentation/pull/3597

**Pull request merged**